src/text/parsers/timestamp.rs

```diff
@@ -32,93 +33,88 @@ pub(crate) fn parse_timestamp(input: &str) -> IResult<&str, TextValue> {
 
 /// Matches the text representation of a timestamp value with year precision (e.g. `2018T`) and
 /// returns the resulting Timestamp as a [TextValue::Timestamp].
-fn timestamp_precision_y(input: &str) -> IResult<&str, TextValue> {
-    map_res::<_, _, _, _, IonError, _, _>(
```

Comment:
🗺️ This module has (lots of) the same kinds of changes we've already seen in the other modules. Namely: replacing `map`/`map_res` with the two-phase approach, bubbling up failures, and dealing with `Ok((remaining_input, value))` success tuples with `.1`.